instrumentation/instrument.go

```diff
@@ -182,79 +179,23 @@ func InstrumentHandlerFuncWithOpts(opts prometheus.Opts, handlerFunc func(http.R
 		f.Inc()
 		defer f.Dec()
 
-		delegate := &responseWriterDelegator{ResponseWriter: w}
-		//out := computeApproximateRequestSize(r)
+		stats := stats{}
 
-		_, cn := w.(http.CloseNotifier)
-		_, fl := w.(http.Flusher)
-		_, hj := w.(http.Hijacker)
-		_, rf := w.(io.ReaderFrom)
-		var rw http.ResponseWriter
-		if cn && fl && hj && rf {
```

Comment:
This check here is to prevent creating delegator with all methods if not implemented by the base `ReponseWrapper`.
Removing it will cause panic when the base `ResponseWriter` does not conform `http.Flusher`